apps/common/utils/toWagmiProvider.tsx

```diff
@@ -23,18 +24,14 @@ export async function toWagmiProvider(connector: Connector): Promise<TWagmiProvi
 	});
 }
 
-// PrepareWriteContractResult.request.value should be bigint instead of undefined
-
-export async function writeContract(config: PrepareWriteContractResult): Promise<TTxResponse> {
-	try {
-		const {hash} = await wagmiWriteContract(config.request);
-		const receipt = await waitForTransaction({chainId: config.request.chainId, hash});
-		return ({isSuccessful: receipt.status === 'success', receipt});
-	} catch (error) {
-		console.error(error);
-		const errorAsBaseError = error as BaseError;
-		captureException(errorAsBaseError);
-		return ({isSuccessful: false, error: errorAsBaseError || ''});
-	}
+export type TWriteTransaction = {
+	connector: Connector;
+	contractAddress: TAddressWagmi;
+	statusHandler?: (status: typeof defaultTxStatus) => void;
+}
 
+export function assertAddress(addr: string): asserts addr is TAddress {
+	assert(isTAddress(addr), 'Address is not an address');
```

Comment:
nit: Sounds a bit strange "Address is not an address" :)

```suggestion
	assert(isTAddress(addr), 'Address provided is invalid');
```